### PR TITLE
Add Cosmos Reason 2 - Worker Safety Recipe OSS

### PR DESCRIPTION
**Pull request by paularamo**
## Description

This recipe demonstrates a complete video reasoning pipeline for automating industrial safety inspections in challenging **"brownfield"** environments using [**Cosmos Reason 2**](https://github.com/nvidia-cosmos/cosmos-reason2) 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement

## Changes Made

- [x] Added/updated documentation
- [x] Added/updated examples
- [ ] Fixed bugs or issues
- [ ] Improved code quality
- [ ] Updated dependencies

## Testing

- [x] I have tested the changes locally
- [X] Documentation builds successfully
- [X] Pre-commit hooks pass
- [X] Examples run without errors
- [X] Links and references are valid

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

## Additional Notes

Includes embedded media (webm clips) and links to the setup guide and main notebook for this recipe. 

Please let me know your feedback about this recipe. 

Thanks Cosmos Team.

### CODE REVIEW

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+## 1. Loading the Dataset
+
+We start by loading the dataset from the Hugging Face Hub into FiftyOne. This dataset contains Full HD (1920x1080) clips at 24 fps, ranging from 1 to 20 seconds in length. 
+
+```python
+import fiftyone as fo
+import fiftyone.utils.huggingface as fouh
+
+# Load the dataset (persistent=True ensures we don't re-download on every run)
+dataset = fo.load_dataset("pjramg/Safe_Unsafe_Test")
```
[the remark this answers is not in the document]
Comment (paularamo) [reply docs/recipes/inference/reason2/worker_safety/inference.md:81]: This is better described in the notebook in the Scripts folder

File: scripts/examples/reason2/worker_safety.ipynb [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+      "cell_type": "markdown",
+      "id": "17b45872",
+      "metadata": {},
+      "source": [
+        "### 5. Initialize the model and create the prompts"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
```
[the remark this answers is not in the document]
Comment (paularamo) [reply scripts/examples/reason2/worker_safety.ipynb]: There is a note in the markdown cell that clarify the way devs should run the inference_sample.py file

File: README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ A comprehensive guide for working with the **NVIDIA Cosmos ecosystem**—a suite
 
 | **Date** | **Recipe** | **Model** | **Description** |
 |----------|------------|-----------|-----------------|
+| Jan 30 | [Worker Safety in a Classical Warehouse](recipes/inference/reason2/worker_safety/inference.md) | Cosmos Reason 2 | Demonstrates a complete video reasoning pipeline for automating industrial safety inspections in challenging "brownfield" environments using |
```
[the remark this answers is not in the document]
Comment (paularamo) [reply README.md]: Waiting for @jingyijin2 feedback on this.

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+---
+
+## Pipeline Overview
+
+This is the end-to-end flow:
+
+1. Load data: ingest the [`Safe_Unsafe_Test`](https://huggingface.co/datasets/pjramg/Safe_Unsafe_Test) dataset from Hugging Face.
+2. Define prompts: encode the dataset's specific visual rules into system and user prompts.
+3. Run inference: process videos using [`Cosmos-Reason2-2B`](https://github.com/nvidia-cosmos/cosmos-reason2) via Hugging Face Transformers.
```
[the remark this answers is not in the document]
Comment (paularamo) [reply docs/recipes/inference/reason2/worker_safety/inference.md]: Adding this in the next commit.

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+
+User prompt (the strict classification table): we provide the model with the exact definitions from Table 1 of the dataset paper:
+
+| ID | Label | Visual Definition (Ground Truth) |
+| :--- | :--- | :--- |
+| 0 | Safe Walkway Violation | Walking OUTSIDE the Green Path. |
+| 4 | Safe Walkway | Walking INSIDE the Green Path. |
+| 1 | Unauthorized Intervention | Interacting with machine board WITHOUT a green vest. |
+| 3 | Forklift Overload | Carrying 3 or more blocks. |
```
[the remark this answers is not in the document]
Comment (paularamo) [reply docs/recipes/inference/reason2/worker_safety/inference.md]: Applying this in the next commit.

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+> **Authors:** [Paula Ramos, PhD](https://www.linkedin.com/in/paula-ramos-phd/)
+> **Organization:** [NVIDIA](https://nvidia.com/)
+
+## Overview
+
+This recipe demonstrates a complete video reasoning pipeline for automating industrial safety inspections in challenging **"brownfield"** environments using [**Cosmos Reason 2**](https://github.com/nvidia-cosmos/cosmos-reason2) and [FiftyOne](https://github.com/voxel51/fiftyone).
+
+It shows how to prompt a Video Language Model (VLM) to ignore environmental noise (like faded paint or old machinery) and strictly classify worker behaviors based on visual ground truths defined in the Video Dataset for Safe and Unsafe Behaviours.
+
+Main notebook: [Worker Safety notebook](worker_safety.ipynb)
```
Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/inference.md:12]: the notebook file is missing in this PR. Could you please add?

File: docs/recipes/inference/reason2/worker_safety/worker_safety.py [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+# ### 5. Initialize the model and create the prompts
+
+# %%
+# #!/usr/bin/env -S uv run --script
+import json
+import warnings
+from pathlib import Path
+import torch
+import transformers
+import fiftyone as fo
```
Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/worker_safety.py]: duplicated import?

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+
+---
+
+## Setup
+
+Before running this recipe, ensure you have the necessary environment configuration.
+
+Prerequisites:
+
+- A GPU environment (e.g., NVIDIA RTX PRO 5000 Blackwell) with CUDA 13.0 available.
```
Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/inference.md]: This GPU model should not be a pre-requisite, but rather a statement that this recipe was created using that GPU. So how about converting it to:
"- An NVIDIA GPU with CUDA support (this recipe was tested on an NVIDIA RTX PRO 5000 Blackwell GPU with CUDA 13.0)."

File: docs/recipes/inference/reason2/worker_safety/setup.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+git lfs install
+```
+
+Verify your GPU + NVIDIA driver + CUDA support:
+
+```bash
+nvidia-smi
+```
+
+> In this setup, the machine is a **laptop with an NVIDIA RTX PRO 5000 Blackwell GPU** running **Ubuntu 24.04** and supporting **CUDA 13.0**, so we use the **cu130** environment extra.
```
Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/setup.md]: similarly, how about rewording it to:
"> **Note:** This recipe was tested on a laptop with an **NVIDIA RTX PRO 5000 Blackwell GPU** running **Ubuntu 24.04** and supporting **CUDA 13.0**. For this configuration, we use the **cu130** environment extra. You may need to adjust the CUDA version based on your system. "

File: docs/recipes/inference/reason2/worker_safety/inference.md [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+
+This approach can be generalized to:
+
+- Construction site monitoring (PPE detection).
+- Retail loss prevention.
+- Logistics and inventory auditing.
+
+For the full code and to run this analysis yourself, verify you have the `pjramg/Safe_Unsafe_Test` dataset and the Cosmos-Reason2 model accessible in your environment.
+
+Run the full workflow in the main notebook: [Worker Safety notebook](worker_safety.ipynb).
```
Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/inference.md:266]: Add a reference to the dataset?
```
### References

- Safe and Unsafe Behaviours Dataset: [Mendeley Data](https://data.mendeley.com/datasets/xjmtb22pff/1)
- Original Paper: Fernandes, P., et al. (2024). "Video Dataset for Safe and Unsafe Behaviours Detection in Industrial Environments." *Data in Brief*. [DOI: 10.1016/j.dib.2024.111258](https://www.sciencedirect.com/science/article/pii/S235234092400756X)
```

Comment (jingyijin2) [inline docs/recipes/inference/reason2/worker_safety/inference.md:266]: Help add date and the citation:
```
---

## Document Information

**Publication Date:** February 4, 2026

### Citation

If you use this recipe or reference this work, please cite it as:

```bibtex
@misc{cosmos_cookbook_worker_safety_2026,
  title={Worker Safety in a Classical Warehouse with Cosmos Reason 2},
  author={Ramos, Paula},
  organization={NVIDIA},
  year={2026},
  month={February},
  howpublished={\url{https://nvidia-cosmos.github.io/cosmos-cookbook/recipes/inference/reason2/worker_safety/inference.html}},
  note={NVIDIA Cosmos Cookbook}
}
```

**Suggested text citation:**

> Paula Ramos (2026). Worker Safety in a Classical Warehouse with Cosmos Reason 2. In *NVIDIA Cosmos Cookbook*. NVIDIA. Accessible at <https://nvidia-cosmos.github.io/cosmos-cookbook/recipes/inference/reason2/worker_safety/inference.html>
```

File: README.md (hunk 3 above)
Comment (jingyijin2) [inline README.md]: Suggest replacing with
```
| Feb 3 | [Worker Safety in a Classical Warehouse](recipes/inference/reason2/worker_safety/inference.md) | Cosmos Reason 2 | Zero-shot industrial safety compliance and hazard detection in classical warehouse environments using context-aware prompt engineering |
```
updated date, and make the overall wording more consistent in style.

File: README.md
Comment (jingyijin2) [inline README.md]: Help remove the last row. We only keep 6 most recent recipes.

**Review by jingyijin2 — COMMENTED** [review]
Great work! Thank you, @paularamo !

File: docs/recipes/inference/reason2/worker_safety/inference.md (hunk 6 above)
Comment (paularamo) [reply docs/recipes/inference/reason2/worker_safety/inference.md:12]: I will do that. The gitignore file is not allowing me to add notebooks to the PR. I will force using ```git add -f docs/recipes/inference/reason2/worker_safety/worker_safety.ipynb```

File: scripts/examples/reason2/worker_safety.ipynb [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+            "  }\n",
+            "}\n",
+            "```\n",
+            "--------------------\n",
+            "[W122 12:18:07.806639126 AllocatorConfig.cpp:28] Warning: PYTORCH_CUDA_ALLOC_CONF is deprecated, use PYTORCH_ALLOC_CONF instead (function operator())\n"
+          ]
+        }
+      ],
+      "source": [
+        "!python ../scripts/inference_sample.py"
```
[the remark this answers is not in the document]
Comment (paularamo) [reply scripts/examples/reason2/worker_safety.ipynb]: There is an explanation in the cell above

File: docs/index.md [hunk 12]
```diff
@@ -21,6 +21,7 @@ The Cosmos ecosystem supports the complete Physical AI development lifecycle--fr
 
 | **Date** | **Recipe** | **Model** |
 |----------|------------|-----------|
+| Jan 30 | [Worker Safety in a Classical Warehouse](recipes/inference/reason2/worker_safety/inference.md) | Cosmos Reason 2 |
```
[the remark this answers is not in the document]
Comment (paularamo) [reply docs/index.md:24]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/nvidia-cosmos/cosmos-cookbook/pull/162#discussion_r2765719105)
